### ISSUE TITLE
`pixi run "any_task_name"`  triggers installing project dependencies

### ISSUE THREAD

**Issue by Okroshiashvili** [prefix-dev/pixi#2429]
### Checks

- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

All my projects, that use pixi has this problem. So, most probably the issue should be same for all others as well.
```shell

```

### Issue description

Today I've updated pixi to 0.35.0 version and got this issue. Every time I run any task, pixi tries to install project dependencies again and only after that runs the task. After downgrading to previous version, 0.34.0 solved the issue.

### Expected behavior

Do not re-install project dependencies if they are already installed and environment exists.

**Comment by SHEscher** [comment]
I have the same issue:

I have dependencies in my `pyproject.toml` :

```toml
# pyproject.toml
...

dependencies = [
    "numpy",
    "pandas",
    ...
 ]

...
```

I run this (dummy) command with the verbose flag:

```shell
pixi run -v python -m MYMODULE
```

Even after running this more than once, I get the following output, where pixi (re-)installs the "distribution":

```shell
...
 INFO pixi::install_pypi: Re-install: []
 INFO pixi::install_pypi: Remove: []
 INFO pixi::install_pypi: Prepared 1 package in 3ms
 INFO pixi::install_pypi: Installed 35 packages in 2.45s
```

The issue did not occur with the previous version of `pixi < 0.35.0`.

**Comment by ruben-arts** [comment]
This was fixed in #2587
